spanner_dbapi/connection.py

```diff
@@ -5,131 +5,98 @@
 # https://developers.google.com/open-source/licenses/bsd
 
 from collections import namedtuple
-
-from google.cloud import spanner_v1 as spanner
+from functools import wraps
 
 from .cursor import Cursor
-from .exceptions import InterfaceError
+from .exceptions import InterfaceError, Warning
+from .enums import AutocommitDMLModes, TransactionModes
 
 ColumnDetails = namedtuple("column_details", ["null_ok", "spanner_type"])
 
 
-class Connection:
-    def __init__(self, db_handle):
-        self._dbhandle = db_handle
-        self._closed = False
-        self._ddl_statements = []
+def _connection_closed_check(func):
+    """Raise an exception if attempting to use an already closed connection."""
 
-    def cursor(self):
-        self.__raise_if_already_closed()
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        if self._is_closed:
+            raise InterfaceError("connection is already closed")
+        return func(self, *args, **kwargs)
 
-        return Cursor(self)
+    return wrapped
```

Comment:
After all, it seems that the `@_connection_closed_check` can sensibly be used only once, see the comments below. Therefore we can safely remove this helper and replace the decorator with a simple two-liner check in the sole method that actually uses it.